src/vulkan/pipeline.cc

```diff
@@ -172,6 +172,45 @@ Result Pipeline::CreateDescriptorSets() {
   return {};
 }
 
+VkPushConstantRange Pipeline::GetPushConstantRange() {
+  if (push_constant_data_.empty())
+    return {};
+
+  auto it =
+      std::min_element(push_constant_data_.begin(), push_constant_data_.end(),
+                       [](const BufferData& a, const BufferData& b) {
+                         return a.offset < b.offset;
+                       });
+  if (it == push_constant_data_.end())
+    return {};
+
+  uint32_t first_offset = it->offset;
+
+  it = std::max_element(
+      push_constant_data_.begin(), push_constant_data_.end(),
+      [](const BufferData& a, const BufferData& b) {
+        return a.offset + static_cast<uint32_t>(a.size_in_bytes) <
+               b.offset + static_cast<uint32_t>(b.size_in_bytes);
+      });
+  if (it == push_constant_data_.end())
+    return {};
```

Comment:
Same here. Assert the iterator is not the end iterator.

Reply:
done